### DISCUSSION TITLE
Reach Matter device behind IPv4 ISP address without bridge

### DISCUSSION THREAD

**Discussion by amunabv** [arendst/Tasmota#21432] [Support request]
Hey,
I've been fiddling around with ESP32C6, TasMota, Matter. It works and is paired in the GHASFM-3p-ecosystem app to pair and control matter devices without the need for a hub. This all works fine and dandy and I can control the LED light from it.

Experimenting further I wondered if the device would be reachable from over the Internet; turned out it wouldn't. Started thinking about IPv6 support for direct reaching of device and then noticed my provider ISP only gives my modem IPv4 address (damn Odido). So my incredibely cool and awesome ESP32C6 Tasmota device has to sit behind the NAT of modem.

I was wondering if it is possible if I'd somehow tell the Tasmota to include, (during commissioning?), it's public IPv4 gateway address and a specified incoming port, then have the modem/router these Matter messages forwarded to the ESP32C6/Tasmota IPv4 address? Like the way NAT is done in local lan for share 1 modem.

**Comment by s-hadinger** [comment]
Matter is really a LAN protocol. The only option is to have a VPN to the LAN. Also take into account that it relies on local-link ipv6 and MDNS, both protocols don't work well over the internet
